control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs

```diff
@@ -130,61 +135,139 @@ async fn nexus_reconciler(
+#[tracing::instrument(skip(nexus, context), level = "trace", fields(nexus.uuid = %nexus.uuid(), request.reconcile = true))]
+pub(super) async fn faulted_children_remover(
+    nexus: &mut OperationGuardArc<NexusSpec>,
+    child: &ChildUri,
+    context: &PollContext,
+) -> Result<(), SvcError> {
+    let nexus_uuid = nexus.uuid();
+    let nexus_state = context.registry().nexus(nexus_uuid).await?;
+
+    info!(%child, "Attempting to remove faulted child");
+    nexus
+        .remove_child_by_uri(context.registry(), &nexus_state, child, true)
+        .await?;
+
+    info!(%child ,"Successfully removed faulted child");
+    Ok(())
+}
+
+/// Returns true if time elapsed from child fault time is greater or equal to wait time duration.
+async fn is_time_elapsed(
+    fault_time: Option<SystemTime>,
+    wait_time: Option<Duration>,
+    uri: &ChildUri,
+) -> bool {
+    if let Some(fault_time) = fault_time {
+        if let Some(wait_time) = wait_time {
+            if let Ok(elapsed) = fault_time.elapsed() {
+                info!(
+                    child.uri = %uri,
+                    "waited {:?} for child to comeback. Wait time is {:?}",
+                    elapsed, wait_time
+                );
+                return elapsed >= wait_time;
+            };
+        }
+    }
+    false
+}
+
+/// If we can get replica for a child from registry. It means that child node is online, Pool is
+/// imported containing the child/replica.
+async fn get_child_replica(
+    nexus: NexusSpec,
+    child: &Child,
+    context: &PollContext,
+) -> Result<Replica, SvcError> {
+    for nexus_child in nexus.children.iter() {
+        if let Some(replica) = nexus_child.as_replica() {
+            if replica.uri() == &child.uri {
+                return if let Ok(replica) = context.registry().replica(replica.uuid()).await {
+                    Ok(replica)
+                } else {
+                    Err(SvcError::ReplicaNotFound {
+                        replica_id: replica.uuid().clone(),
+                    })
+                };
+            } else {
+                continue;
+            }
+        }
+    }
+    Err(SvcError::NotFound {
+        kind: ResourceKind::Node,
+        id: "Child replica not found".to_string(),
+    })
+}
+
+/// Tries to mark child as Online. Returns error on failing to do so.
+pub(super) async fn initialize_partial_rebuild(
+    nexus: &NexusId,
+    child: &ChildUri,
+    context: &PollContext,
+) -> Result<(), SvcError> {
+    let nexus_state = context.registry().nexus(nexus).await?;
+    info!(%child, "Making online of {:?} nexus", nexus);
+    let node = context.registry().node_wrapper(&nexus_state.node).await?;
+    let online_context = NexusChildActionContext::new(&nexus_state.node, &nexus.clone(), child);
+    let ctx = NexusChildActionContextNode::new(online_context, context.registry());
+    // TODO: Check if we can handle things differently for different SvcError.
+    node.online_child(ctx).await?;
```

Comment:
need to handle return failures here as you were doing in earlier version of change.

Reply:
Its handled, In the caller end if this returns err, We start Full rebuild of the child.